src/main/java/attendance/domain/AttendanceStatus.java

```diff
@@ -0,0 +1,75 @@
+package attendance.domain;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.List;
+
+import static attendance.domain.AttendanceRiskType.NONE;
+import static attendance.domain.AttendanceRiskType.find;
+import static attendance.domain.AttendanceStateType.ABSENCE;
+import static attendance.domain.AttendanceStateType.LATE;
+import static attendance.domain.AttendanceStateType.values;
+
+public class AttendanceStatus implements Comparable<AttendanceStatus> {
+
+    private final EnumMap<AttendanceStateType, Integer> status;
+    private final AttendanceRiskType riskType;
+
+    public AttendanceStatus(final List<Attendance> attendances) {
+        status = calculateAttendanceStatus(attendances);
+        riskType = calculateAttendanceRisk();
+    }
+
+    public static AttendanceStatus of(final List<Attendance> attendances) {
+        return new AttendanceStatus(attendances);
+    }
+
+    public boolean isNotNoneState() {
+        return riskType != NONE;
+    }
+
+    @Override
+    public int compareTo(final AttendanceStatus other) {
+        if (riskType == other.riskType) {
+            int thisScore = calculateScore();
+            int otherScore = other.calculateScore();
+
+            return Integer.compare(otherScore, thisScore);
+        }
+        return other.riskType.compareTo(this.riskType);
+    }
+
+    public EnumMap<AttendanceStateType, Integer> getStatus() {
+        return status;
+    }
+
+    public AttendanceRiskType getRiskType() {
+        return riskType;
+    }
+
+    private EnumMap<AttendanceStateType, Integer> calculateAttendanceStatus(final List<Attendance> attendances) {
+        EnumMap<AttendanceStateType, Integer> status = new EnumMap<>(AttendanceStateType.class);
+        Arrays.stream(values())
+                .forEach(stateType -> {
+                    int stateCount = calculateStateCount(attendances, stateType);
+                    status.put(stateType, stateCount);
+                });
+        return status;
```

Comment:
```suggestion
        return Arrays.stream(AttendanceStateType.values())
                .collect(Collectors.toMap(
                        stateType -> stateType,
                        stateType -> calculateStateCount(attendances, stateType),
                        (a, b) -> b,
                        () -> new EnumMap<>(AttendanceStateType.class)
                ));
```
로 표현될 수 있겠네요. 

stream의 foreach 종단 연산은 안티패턴입니다.

https://tecoble.techcourse.co.kr/post/2020-05-14-foreach-vs-forloop/

Reply:
전혀 모르고 있었던,, 사실입니다. 리뷰 감사합니다! ☺️